fix: remove /src/ from .npmignore (for sourcemaps)

justingrant opened the pull request:
Crrently, SDKV3 publishes sourcemaps to npm but doesn't publish the source files that those sourcemaps point to. This missing source files breaks debugging use-cases, especially for VSCode users because the VSCode debugger relies on source files for setting breakpoints in the debugger, for debugger call stacks, for "step into" original source, and any other debugger use-cases.  Even outside of debugging use-cases, it's helpful for developers consuming transpiled libraries to have original source so they can better understand what the library is doing. Finally, some tools will show warnings when sourcemaps are present but source files are missing, and this PR will fix these warnings.

This PR:
* removes `/src` from npm for non-client packages, but adds adds `jest.config.js` and `*.spec.ts` to these packages' .npmignore because it's not expected to be able to run your tests using an NPM download.
* removes `*.ts` from npm for client packages. (also removes `!*.d.ts` which is now unnecessary)
* in 10 non-client packages, changes `sourceRoot` to `rootDir` in tsconfig.json.  The latter setting is used by all other SDKV3 packages, and it produces sourcemaps with an empty `sourceRoot` field. This is ideal, because the `sourceRoot` feature turned out to be buggy in some sourcemap-reading and sourcemap-writing tools. Leaving it blank maximizes compatibility with more tools.
 
This PR is similar to https://github.com/aws/aws-sdk-js-crypto-helpers/pull/5 in the AWS Crypto Helpers library, and https://github.com/aws-amplify/amplify-js/pull/2680 which made similar changes to AWS Amplify library earlier this year.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

aws-sdk-js-automation commented:
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://zr2fgudy0e.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=99671f9e-dacf-4b4d-9e0c-9eac41ae01eb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

aws-sdk-js-automation commented:
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://zr2fgudy0e.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=fbcd4d49-99fe-4175-a461-6a347ee8125b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

justingrant commented:
After my initial commit last week, I made a few additional changes: 
* omitted test-only files from npm
* fixed a sourcemap-configuration issue with 10 non-client projects.

I updated the initial comment to reflect these additional changes.

trivikr commented:
Thank you @justingrant for posting this PR.
Closing as we no longer exclude TypeScript files in alpha clients

The code is currently in `smithy-codegen` branch, and would be merged to master when it's stable.

https://github.com/aws/aws-sdk-js-v3/blob/dc37f60b0f225ba5fd2fe9dff204ceadab2dc81d/clients/client-acm/.npmignore#L1-L4

Verified as follows:
```console
$ yarn add @aws-sdk/client-acm@alpha
$ yarn list @aws-sdk/client-acm
└─ @aws-sdk/client-acm@1.0.0-alpha.12
$ ls node_modules/@aws-sdk/client-acm 
ACM.ts                          node_modules
ACMClient.ts                    package.json
CHANGELOG.md                    protocols
LICENSE                         runtimeConfig.browser.ts
README.md                       runtimeConfig.shared.ts
commands                        runtimeConfig.ts
dist                            tsconfig.es.json
endpoints.ts                    tsconfig.json
index.ts                        types
models
```
